src/nfagraph/ng_misc_opt.cpp

```diff
@@ -385,8 +385,7 @@ bool improveGraph(NGHolder &g, som_type som) {
 
     const vector<NFAVertex> ordering = getTopoOrdering(g);
 
-    return enlargeCyclicCR(g, som, ordering)
-        | enlargeCyclicCR_rev(g, ordering);
+    return enlargeCyclicCR(g, som, ordering) || enlargeCyclicCR_rev(g, ordering);
```

Comment:
This is not equivalent, FYI. Graph g can be changed in both directions with `|` rather than `||`

Reply:
clang 14 complains if you use bitwise OR on booleans, hence the change.

Reply:
unless you mean for || the compiler only has to evaluate the first to be true for optimization, so the second may not be executed. In which case, I will probably have to evaluate both separately and just OR the results.

Reply:
Yes, I think that's what @danlark1 meant: both sides may need to be evaluated for their side effects, so switching from `|` to `||` could be incorrect.

It would be safer to rewrite this instead like you mentioned, ORing the results:
```suggestion
    bool lhs = enlargeCyclicCR(g, som, ordering);
    bool rhs = enlargeCyclicCR_rev(g, ordering);
    return lhs || rhs;
```